src/syft/lib/pandas/__init__.py

```diff
@@ -0,0 +1,199 @@
+        ("pandas.Series.eq", "pandas.Series"),
+        ("pandas.Series.argsort", "pandas.Series"),
+        ("pandas.Series.round", "pandas.Series"),
+        ("pandas.Series.head", "pandas.Series"),
+        ("pandas.Series.tail", "pandas.Series"),
+        ("pandas.Series.any", "syft.lib.python.Bool"),
+        ("pandas.Series.shape", "syft.lib.python.Tuple"),
+        ("pandas.Series.all", "syft.lib.python.Bool"),
+        ("pandas.Series.argmax", "syft.lib.python.Int"),
+        ("pandas.Series.nbytes", "syft.lib.python.Int"),
+        ("pandas.Series.mean", "syft.lib.python.Float"),
+        ("pandas.Series.ndim", "syft.lib.python.Int"),
+        ("pandas.Series.size", "syft.lib.python.Int"),
+        ("pandas.Series.hasnans", "syft.lib.python.Bool"),
+        ("pandas.Series.empty", "syft.lib.python.Bool"),
+        ("pandas.Series.T", "pandas.Series"),
+        ("pandas.Series.dropna", "pandas.Series"),
+        ("pandas.Series.to_frame", "pandas.DataFrame"),
+        ("pandas.Series.to_list", "syft.lib.python.List"),
+        (
+            "pandas.Series.sum",
+            UnionGenerator["syft.lib.python.Float", "syft.lib.python.Int"],
+        ),
+        ("pandas.Series.median", "syft.lib.python.Float"),
+        (
+            "pandas.Series.max",
+            UnionGenerator[
+                "syft.lib.python.Bool", "syft.lib.python.Float", "syft.lib.python.Int"
+            ],
+        ),
+        (
+            "pandas.Series.min",
+            UnionGenerator[
+                "syft.lib.python.Bool", "syft.lib.python.Float", "syft.lib.python.Int"
+            ],
+        ),
+    ]
+
+    add_modules(ast, modules)
+    add_classes(ast, classes)
+    add_methods(ast, methods)
+
+    for klass in ast.classes:
+        klass.create_pointer_class()
+        klass.create_send_method()
+        # TODO: Pandas can't have tags and description because they break the dict
+        # klass.create_storable_object_attr_convenience_methods()
+
+    return ast
+
+
+# we cant create Unions that refer to the package itself until the create_ast
+# has completed first so we can call again into post_update_ast to finish these
+# TODO: add support for self referential unions using some kind of post update
+# Issue: https://github.com/OpenMined/PySyft/issues/5323
+def post_create_ast(ast: Globals) -> Globals:
+    self_referencing_methods = [
+        ("pandas.Series.loc", UnionGenerator["pandas.DataFrame", "pandas.Series"]),
+        ("pandas.Series.iloc", UnionGenerator["pandas.DataFrame", "pandas.Series"]),
+        ("pandas.DataFrame.loc", UnionGenerator["pandas.DataFrame", "pandas.Series"]),
+        ("pandas.DataFrame.iloc", UnionGenerator["pandas.DataFrame", "pandas.Series"]),
+    ]
+
+    add_methods(ast, self_referencing_methods)
+
+    return ast
+
+
+update_ast = functools.partial(generic_update_ast, LIB_NAME, create_ast)
+# post_update_ast = functools.partial(generic_update_ast, LIB_NAME, post_create_ast)
```

Comment:
Q: This needs to be removed?

Reply:
Yeah I could remove it until we implement it properly for now it just turned off since its going to need some more solid thought. I took a crack but it seems that doing a post create ast is more complicated than I thought.